packages/mobile/src/paymentRequest/utils.ts

```diff
@@ -0,0 +1,24 @@
+import { PaymentRequest } from 'src/account'
+import { NumberToRecipient, Recipient, RecipientKind } from 'src/recipients/recipient'
+
+export function getRecipientFromPaymentRequest(
+  paymentRequest: PaymentRequest,
+  recipientCache: NumberToRecipient
+): Recipient {
+  const cachedRecipient = paymentRequest.requesterE164Number
+    ? recipientCache[paymentRequest.requesterE164Number]
+    : null
+
+  if (cachedRecipient) {
+    return {
+      ...cachedRecipient,
+      address: paymentRequest.requesterAddress,
+    }
+  } else {
+    return {
+      kind: RecipientKind.Address,
+      address: paymentRequest.requesterAddress,
+      displayName: paymentRequest.requesterAddress,
```

Comment:
How about using requester's phone number (if available, any request type other than request by address), and address if not (Request by address) for `displayName`?

Reply:
great id! Done.